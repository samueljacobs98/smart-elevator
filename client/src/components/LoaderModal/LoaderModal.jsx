import "./LoaderModal.scss";
import loader from "../../assets/gifs/loader.gif";

const LoaderModal = () => {
  return (
    <div className="loader-modal">
      <img className="loader-modal__loader" src={loader} alt="" />
    </div>
  );
};

export default LoaderModal;
