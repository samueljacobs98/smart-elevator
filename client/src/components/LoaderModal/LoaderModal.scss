@use "../../assets/scss/variables" as *;

.loader-modal {
  position: fixed;
  width: 100%;
  height: 100%;
  background-color: rgba($colour-grey-dark, 0.5);
  z-index: 9999;
  display: grid;
  place-items: center;

  &__loader {
    width: 4rem;
    height: 4rem;
  }
}
