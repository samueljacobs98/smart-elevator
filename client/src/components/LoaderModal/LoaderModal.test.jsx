import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import loader from "../../assets/gifs/loader.gif";
import LoaderModal from "./LoaderModal";

describe("LoaderModal", () => {
  const renderComponent = () => render(<LoaderModal />);

  it("renders without crashing", () => {
    renderComponent();

    expect(screen).toBeDefined();
  });

  it("renders the loader image", () => {
    renderComponent();

    const loaderImage = screen.getByRole("img");

    expect(loaderImage).toHaveAttribute("src", loader);
  });

  it("renders the loader image with an alt text", () => {
    renderComponent();

    const loaderImage = screen.getByRole("img");

    expect(loaderImage).toHaveAttribute("alt", "Loading animation");
  });

  it("should match snapshot", () => {
    const { container } = renderComponent();

    expect(container).toMatchSnapshot();
  });
});
