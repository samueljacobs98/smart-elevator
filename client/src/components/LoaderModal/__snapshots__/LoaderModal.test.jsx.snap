// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoaderModal should match snapshot 1`] = `
<div>
  <div
    class="loader-modal"
  >
    <img
      alt="Loading animation"
      class="loader-modal__loader"
      src="loader.gif"
    />
  </div>
</div>
`;
